cni/pkg/repair/netns.go

```diff
@@ -104,27 +97,26 @@ func getPodNetNs(pod *corev1.Pod) (string, error) {
 	}
 	oldest := uint64(math.MaxUint64)
 	best := ""
-	// We will iterate over all processes. Our goal is to find a process with the same network ID as we found above.
+
+	// We will iterate over all processes. Our goal is to find a process whose namespace has a veth with an IP matching the pod.
 	// There should be 1 or 2 processes that match: the pause container should always be there, and the istio-validation *might*.
 	// We want the pause container, as the istio-validation one may exit before we are done.
 	// We do this by detecting the longest running process. We could look at `cmdline`, but is likely more reliable to weird platforms.
 	for _, p := range procs {
+		match := false
 		ns := getPidNamespace(p.PID)
-		fd, err := unix.Open(ns, unix.O_RDONLY, 0)
-		if err != nil {
-			// Not uncommon, many processes are transient and we have a TOCTOU here.
-			// No problem, must not be the one we are after.
-			log.Debugf("failed to open pid %v: %v", p.PID, err)
-			continue
-		}
-		id, err := netlink.GetNetNsIdByFd(fd)
-		_ = unix.Close(fd)
+
+		err := netns.WithNetNSPath(ns, func(_ netns.NetNS) error {
```

Comment:
I am worried this is way too slow. On my machine with 2000 processes it takes 1s to do this traversal

Reply:
And it spams a warning log about 'failed to check proc' on each one

Reply:
Hmm, the log should be debug, yeah.  Another idea to speed it up would be to track which namespaces we have already checked the interfaces of and skip doing the WithNetNSPath call for those namespaces.  That should help quite a bit I would presume.